### Numpy

#### NumPy is a library for the Python programming language, adding support for large, multi-dimensional arrays and matrices, along with a large collection of high-level mathematical functions to operate on these arrays.

#### Please complete Pandas Tutorial before starting on Numpy.

Datasets Required:
1. exchange-rates-sgd-per-unit-of-usd-daily.csv

#### Import pandas and numpy

In [2]:
import pandas as pd
import numpy as np

#### Import CSV file into a dataframe

In [3]:
df = pd.read_csv('exchange-rates-sgd-per-unit-of-usd-daily.csv')
df.head()

date  exchange_rate_usd
0  1988-01-08             2.0443
1  1988-01-15             2.0313
2  1988-01-22             2.0205
3  1988-01-29             2.0182
4  1988-02-05             2.0160

### Quartiles

#### The first, second and third quartiles of a series, are the 25 percentile, 50 percentile and 75 percentile respectively.

Use np.percentile(SERIES, `x`) to get the `x`th percentile of the distribution.

Find the 25th. 50th and 75th percentile of the exchange_rate_usd.

In [4]:
print(np.percentile(df['exchange_rate_usd'], 25))
print('-------------')
print(np.percentile(df['exchange_rate_usd'], 50))
print('-------------')
print(np.percentile(df['exchange_rate_usd'], 75))


1.3033
-------------
1.4622
-------------
1.6691


### Interquartile Range

Interquartile range is defined as `third quartile - first quartile`.

This can be used to determine outliers, which are values that lie outside `[ (1st_quartile - 1.5 * IQR) , (3rd_quartile + 1.5 * IQR) ]`

Find the interquartile range of exchange_rate_usd.

Find the lower and upper bound of the exchange_rate_usd.

Find all days which are outliers.

Hint* Set conditions to identify outliers, and apply the conditions to the dataframe.